File: 2020-09-12 - Complejidad algorítmica/0xf2 - Partir un arreglo en 2/quadratic.cpp
#include <vector>
#include <iostream>
using namespace std;


int main() {
  int n;
  cin >> n;
  vector<int> a(n);
  for (int i = 0; i < n; ++i) {
    cin >> a[i];
  }

  for (int r = 0; r < n; ++r) {
    int left = 0, right = 0;
    for (int i = 0; i < r; ++i) {
      left += a[i];
    }
    for (int i = r; i < n; ++i) {
      right += a[i];
    }
    if (left > 0 and right < 0) {
      cout << r << endl;
      return 0;
    }
  }
  cout << "Impossible" << endl;
  return 0;
}

File: 2020-09-12 - Complejidad algorítmica/0xf2 - Partir un arreglo en 2/quadratic.py
def solve(a):
  n = len(a)
  for i in range(n):
    left = sum(a[0:i])
    right = sum(a[i:n])
    if left > 0 and right < 0:
      print(i)
      return
  print("Impossible")

if __name__ == "__main__":
  n = int(input())
  a = [int(x) for x in input().split()]
  solve(a)